src/main/kotlin/org/opensearch/reportsscheduler/action/GetAllReportDefinitionsAction.kt

```diff
@@ -39,6 +41,6 @@ internal class GetAllReportDefinitionsAction @Inject constructor(
      * {@inheritDoc}
      */
     override fun executeRequest(request: GetAllReportDefinitionsRequest, user: User?): GetAllReportDefinitionsResponse {
-        return ReportDefinitionActions.getAll(request, user)
+        return ReportDefinitionActions.getAll(request, pluginClient, user)
```

Comment:
Should we add null checks or make it a non-null type with initialization validation?
```
  fun executeRequest(request: GetAllReportDefinitionsRequest, user: User?, pluginClient: PluginClient?): GetAllReportDefinitionsResponse {
      val client = pluginClient ?: throw IllegalStateException("PluginClient not initialized")
 ...
  }
  ```